packages/overlay/src/overlay-trigger.ts

```diff
@@ -179,49 +77,78 @@ export class OverlayTrigger extends LitElement {
         `;
     }
 
-    private onClickSlotChange(event: Event): void {
-        /* istanbul ignore if */
-        if (!event.target) {
-            return;
+    public onTriggerClick(): void {
+        /* istanbul ignore else */
+        if (this.targetContent && this.clickContent) {
+            this.closeClickOverlay = Overlay.open(
+                this.targetContent,
+                'click',
+                this.clickContent,
+                {
+                    offset: this.offset,
+                    placement: this.placement,
+                }
+            );
         }
-        const slot = event.target as HTMLSlotElement;
-        const content = this.extractSlotContent(slot);
+    }
 
-        if (content) {
-            this.clickContent = content;
+    public onTriggerMouseEnter(): void {
+        /* istanbul ignore else */
+        if (this.targetContent && this.hoverContent) {
+            this.closeHoverOverlay = Overlay.open(
+                this.targetContent,
+                'hover',
+                this.hoverContent,
+                {
+                    offset: this.offset,
+                    placement: this.placement,
+                }
+            );
         }
     }
 
+    public onTriggerMouseLeave(): void {
+        /* istanbul ignore else */
+        if (this.closeHoverOverlay) {
+            this.closeHoverOverlay();
+            delete this.closeHoverOverlay;
+        }
+    }
+
+    private onClickSlotChange(event: Event): void {
+        const content = this.extractSlotContentFromEvent(event);
+        this.clickContent = content || undefined;
+    }
+
     private onHoverSlotChange(event: Event): void {
+        const content = this.extractSlotContentFromEvent(event);
+        this.hoverContent = content || undefined;
+    }
+
+    private onTargetSlotChange(event: Event): void {
+        const content = this.extractSlotContentFromEvent(event);
+        this.targetContent = content || undefined;
```

Comment:
You fixed the below return, but I was hoping it would allow this and the two previous methods to not have this:
```suggestion
        this.targetContent = content;
```
With `private extractSlotContentFromEvent(event: Event): HTMLElement | undefined {` these are the same?

Reply:
Fixed